--- squashfs-tools/process_fragments.c	2016-08-25 09:06:22.983529595 -0400
+++ squashfs-tools-patched/process_fragments.c	2016-08-25 09:06:03.223530354 -0400
@@ -192,9 +192,10 @@
 
 		res = compressor_uncompress(comp, buffer->data, data, size,
 			block_size, &error);
-		if(res == -1)
-			BAD_ERROR("%s uncompress failed with error code %d\n",
-				comp->name, error);
+        // CJH: Decompression errors are displayed elsewhere
+		//if(res == -1)
+		//	BAD_ERROR("%s uncompress failed with error code %d\n",
+		//		comp->name, error);
 	} else if(compressed_buffer)
 		memcpy(buffer->data, compressed_buffer->data, size);
 	else {
diff --strip-trailing-cr -NBbaur squashfs-tools/read_fs.c squashfs-tools-patched/read_fs.c
